pootle/apps/pootle_app/migrations/0012_set_directory_tp.py

```diff
@@ -0,0 +1,32 @@
+# -*- coding: utf-8 -*-
+# Generated by Django 1.9.10 on 2016-11-02 19:58
+from __future__ import unicode_literals
+
+from django.db import migrations
+
+
+def set_directory_tp(apps, schema_editor):
+    dirs = apps.get_model("pootle_app.Directory").objects.all()
+    TP = apps.get_model("pootle_translationproject.TranslationProject")
+    tps = TP.objects.all()
+    for directory in dirs:
```

Comment:
I would have done it the other way: for each tp find all dirs whose pootle_path startswith tp.pootle_path and assign the tp to all of them in a bulk update. Is there any specific reason why you chose to do it this way (try..except needed)?